File: tests/omp/task_dep4.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->

  !! Omp.task ~clause:[Shared ["x"]; Depend [Out [Dep_var "x"]]] [nbMulti; cWriteVar "x"];
  !! Omp.task ~clause:[Shared ["x"]; Depend [In [Dep_var "x"]]] [nbMulti; cFun "printf"];
     let tg = [cSeq ~args_pred:(target_list_one_st [cWriteVar "x"]) ()] in
  !! Omp.single tg;
  !! Omp.parallel tg;
)

File: tests/omp/task_dep4.cpp
#include <omp.h>
#include <stdio.h>

int main(){
  int x = 1;

  {
    x = 2;
    printf("x + 1 = %d. ", x + 1);
    printf("x + 2 = %d\n ", x + 2);

  }
  return 0;
}